react/src/components/EndpointSelect.tsx

```diff
@@ -56,7 +58,12 @@ const EndpointSelect: React.FC<EndpointSelectProps> = ({
       limit: baiPaginationOption.limit,
       offset: baiPaginationOption.offset,
       filter: baiClient.supports('endpoint-lifecycle-stage-filter')
-        ? `lifecycle_stage == "created"`
+        ? [
+            `(lifecycle_stage == "created")`,
+            searchStr && `(name ilike "%${searchStr}%")`,
+          ]
+            .filter(Boolean)
+            .join(' & ')
```

Comment:
One more condition is needed. You also need to fetch the endpoint information for `controllableValue`. Without this, this component may display `endpoint_id` instead of the endpoint's name if the endpoint not in the first page of query.

Reply:
Currently, the endpoint_list query does not provide filtering based on endpoint_id, but the controllableValue is passing id. Is backend.ai core need to add endpoint_id to the filter options? @yomybaby

Reply:
Another option is to query the endpoint for `value` by adding the `endpoint(endpoint_id: UUID!): Endpoint` query. We can include the existing `EndpointSelectQuery`. Don't forget to use `@skip` to skip the `endpoint` query when the `value` is undefined.